CMakeLists.txt

```
@@ -5,62 +5,90 @@
 
 cmake_minimum_required(VERSION 3.20)
 cmake_policy(SET CMP0057 NEW)
-project(YOMM2 LANGUAGES CXX VERSION 1.5.2)
+
+project(YOMM2 LANGUAGES CXX VERSION 1.6.0)
 
 if(NOT CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 17)
 endif()
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-option(YOMM2_DOWNLOAD_DEPENDENCIES "Set to ON to build missing dependencies from source" OFF)
-include(cmake/download_package.cmake)
+option(YOMM2_SHARED "Build yomm2 as a shared library" OFF)
+option(YOMM2_ENABLE_EXAMPLES "Set to ON to build examples" OFF)
+option(YOMM2_DEBUG_MACROS "Set to ON to debug macros" OFF)
+option(YOMM2_ENABLE_DOC "Set to ON to generate tutorials and reference" OFF)
+option(YOMM2_ENABLE_TESTS "Set to ON to build tests" OFF)
+option(YOMM2_ENABLE_BENCHMARKS "Set to ON to enable benchmarks" OFF)
+option(
+  YOMM2_CHECK_ABI_COMPATIBILITY
+  "Build shared library and examples in different modes" OFF)
 
-# Find Boost dependency
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
-find_package(Boost 1.74 QUIET)
-if(NOT TARGET Boost::headers)
-  if (YOMM2_DOWNLOAD_DEPENDENCIES)
-    download_package(Boost INSTALL_WITH_YOMM)
+include(CMakeDependentOption)
+CMAKE_DEPENDENT_OPTION(YOMM2_ENABLE_BENCHMARKS
+  "Set to ON to build benchmarks" OFF
+  "YOMM2_ENABLE_TESTS" OFF
+)
+
+macro(assign_bool VAR)
+  if(${ARGN})
+    set(${VAR} ON)
   else()
-    message(FATAL_ERROR  "Boost was not found on your system. Set YOMM2_DOWNLOAD_DEPENDENCIES to ON to download and build from the sources.")
+    set(${VAR} OFF)
   endif()
+endmacro()
+
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Due to cross-platform compatibility of Clang, this part could be problematic when having Clang on Windows. For example, Clang can have two frontend on Windows (https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_FRONTEND_VARIANT.html)

A way around could be to check the above variable directly or perhaps to do this ( as per doc on [MSVC flag](https://cmake.org/cmake/help/latest/variable/MSVC.html))
```suggestion
if(NOT MSVC AND CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Related on the topic is this variable https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_SIMULATE_ID.html